src/main/python/covata/delta/utils.py

```diff
@@ -34,3 +34,9 @@ def check_id(arguments):
     return check_arguments(arguments,
                            lambda x: x is not None and str(x) is not "",
                            "must be a nonempty string")
+
+
+def check_metadata(arguments):
+    return check_arguments(arguments,
+                           lambda x: x is not None and dict(x),
```

Comment:
correct - i guess in the case of update_metadata method in api_client, i think it might be right if we allow empty metadata. hmm, something like this might work

```
lambda x: x is not None and len(x) >= 0 if dict(x) else False
```